pre_build_script.py

```diff
@@ -0,0 +1,35 @@
+Import("env")
+
+import platform
+
+extra_macros = {
+    # PP indirection needed
+    'DO_PRAGMA_(x)': '_Pragma(#x)',
+    'DO_PRAGMA(x)': 'DO_PRAGMA_(x)',
+
+    # Useful for disabling warnings on system headers that we don't care about
+    'PUSH_NO_WARNINGS': ''.join([
```

Comment:
just a small one ... what if we rename it to "DISABLE_WARNINGS" and "ENABLE_WARNINGS"? or is there some convention to name it push and pop?

Reply:
We can if you'd like! I was just following the convention from the `#pragma GCC diagnostic push`/`#pragma GCC diagnostic pop` that the macros use (and pushing/popping is a pretty common term for stuff like this)

Reply:
i think it depends on how it works. if you e.g. push 2 times and need at least 2 times pop (stack logic), then we should keep PUSH/POP naming. otherwise enable/disable is more correct

Reply:
Ah I see what you mean, GCC does keep a stack so the `PUSH`/`POP` naming is technically correct ([GCC docs](https://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html)